test/extensions/filters/network/mongo_proxy/config_test.cc

```diff
@@ -67,186 +70,135 @@ TEST(MongoFilterConfigTest, MongoFilterWithEmptyProto) {
   cb(connection);
 }
 
-void handleInvalidConfiguration(const std::string& json_string) {
-  Json::ObjectSharedPtr json_config = Json::Factory::loadFromString(json_string);
-  NiceMock<Server::Configuration::MockFactoryContext> context;
-  MongoProxyFilterConfigFactory factory;
-
-  EXPECT_THROW(factory.createFilterFactory(*json_config, context), Json::Exception);
+void handleInvalidConfiguration(const std::string& yaml_string, const std::string& error_regex) {
+  envoy::config::filter::network::mongo_proxy::v2::MongoProxy config;
+  EXPECT_THROW_WITH_REGEX(TestUtility::loadFromYamlAndValidate(yaml_string, config), EnvoyException,
+                          error_regex);
 }
 
 TEST(MongoFilterConfigTest, InvalidExtraProperty) {
-  std::string json_string = R"EOF(
-  {
-    "stat_prefix": "my_stat_prefix",
-    "access_log" : "path/to/access/log",
-    "test" : "a"
-  }
+  const std::string yaml_string = R"EOF(
+  stat_prefix: my_stat_prefix
+  access_log: path/to/access/log
+  test: a
   )EOF";
 
-  handleInvalidConfiguration(json_string);
+  handleInvalidConfiguration(yaml_string, "test: Cannot find field");
 }
 
-TEST(MongoFilterConfigTest, EmptyConfig) { handleInvalidConfiguration("{}"); }
-
-TEST(MongoFilterConfigTest, InvalidFaultsEmptyConfig) {
-  std::string json_string = R"EOF(
-  {
-    "stat_prefix": "my_stat_prefix",
-    "fault" : {}
-  }
-  )EOF";
-
-  handleInvalidConfiguration(json_string);
+TEST(MongoFilterConfigTest, EmptyConfig) {
+  handleInvalidConfiguration(
+      "{}", R"(StatPrefix: \["value length must be at least " '\\x01' " bytes"\])");
 }
 
-TEST(MongoFilterConfigTest, InvalidFaultsMissingPercentage) {
```

Comment:
Are we losing real coverage here?

Reply:
There's a tricky line here since there was originally a lot of v1 code that tested JSON schema constraints that is now delegated to protoc-gen-validate constraints. In this particular case, the test `InvalidFaultsMissingPercentage` was removed because the absence of a percentage field is no longer invalid, and [implies 100%](https://github.com/envoyproxy/envoy/blob/master/api/envoy/type/percent.proto#L46). I can add a test to confirm that, granted it wouldn't move the code coverage needle since we don't include protobuf-generated code in coverage reports, but I think should be an isolated test specific to the `FractionalPercent` type and not all of its consumers..